### ISSUE TITLE
`Test` method for PredictorInterface

### ISSUE THREAD

**Issue by paxcema** [mindsdb/lightwood#988]
We should add a new method to the predictor interface for arbitrary metric computation given a dataframe with values for the target column.

```python
pdef = ProblemDefinition.from_dict({...})
json_ai = json_ai_from_problem(train_df, problem_definition=pdef)
predictor_class_code = code_from_json_ai(json_ai)
predictor = predictor_from_code(predictor_class_code)
predictor.learn(train_df)

# existing behavior
predictions = predictor.predict(test_df)

# new behavior

# 1) string for already supported metrics and user-defined modules
metrics = predictor.test(test_df, metrics=['mean_absolute_error', 'my_custom_module.method_name'])

# 2) support for callables and lambdas
from sklearn.metrics import mean_absolute_error
metrics = predictor.test(test_df, metrics=[mean_absolute_error, lambda x, y: abs(x - y)])
```

Returned object should be a Pandas dataframe with the value of each passed metric:

```python
In[1]: metrics
Out[1]:
                             metric_name  value
mean_absolute_error  mean_absolute_error  32433
lambda_0                        lambda_0  45576
```

It may also be worth defining an extra kwarg so that the user can retrieve predictions (as if they were using `predict()`):

```python
metrics, predictions = predictor.test(test_df, metrics=['mean_absolute_error'], return_preds=True)
```
